crates/rover-client/src/introspection/schema.rs

```diff
@@ -0,0 +1,295 @@
+                                field_def
+                            }
+                            // [[Type]]!
+                            LIST => {
+                                let type_ref = type_ref.of_type.unwrap();
+                                match type_ref.kind {
+                                    SCALAR | OBJECT | INTERFACE | UNION | ENUM | INPUT_OBJECT => {
+                                        let ty =
+                                            FieldType::new_type(type_ref.name.unwrap(), true, None);
+                                        let list1 = FieldType::new_list(ty, true);
+                                        let list2 = FieldType::new_list(list1, false);
+                                        let mut field_def = Field::new(field.name, list2);
+                                        field_def.description(field.description);
+                                        field_def
+                                    }
+                                    LIST => panic!("Cannot nest more than 2 lists."),
+                                    // [[Type!]]!
+                                    NON_NULL => {
+                                        let type_ref = type_ref.of_type.unwrap();
+                                        match type_ref.kind {
+                                            SCALAR | OBJECT | INTERFACE | UNION | ENUM
+                                            | INPUT_OBJECT => {
+                                                let ty = FieldType::new_type(
+                                                    type_ref.name.unwrap(),
+                                                    false,
+                                                    None,
+                                                );
+                                                let list1 = FieldType::new_list(ty, true);
+                                                let list2 = FieldType::new_list(list1, false);
+                                                let mut field_def = Field::new(field.name, list2);
+                                                field_def.description(field.description);
+                                                field_def
+                                            }
+                                            LIST => panic!("Cannot nest more than 2 lists."),
+                                            ty => panic!("Unknown type: {:?}", ty),
+                                        }
+                                    }
+                                    ty => panic!("Unknown type: {:?}", ty),
+                                }
+                            }
+                            ty => panic!("Unknown type: {:?}", ty),
+                        }
+                    }
+                    ty => panic!("Unknown type: {:?}", ty),
+                }
+            }
+            Other(ty) => panic!("Unknown type: {}", ty),
```

Comment:
I have no idea where `Other` comes from, that's not part of the spec so this may be an artifact of how `graphql-client` works?

Reply:
If you don't know of any other types other than SCALAR,  OBJECT,  INTERFACE, UNION, ENUM, INPUT_OBJECT, LIST and NON_NULL, I take it would be an artefact of graphql_client. I have not been able to cause a panic coming from here, so I am hoping we will never hit this case.

Reply:
That seems right, both according to the spec and looking at the introspection code in `graphql-js`. Maybe `graphql-client` just treats every error as open ended? We do something similar in Apollo iOS to allow for schema evolution.